### PR TITLE
feat: EL and CL URLs as lists

### PR DESCRIPTION
**Pull request by vgorkavenko**
**TODOs:**

- [x] readme

### CODE REVIEW

**Review by dgusakov — APPROVED** [review]

File: src/common/eth-providers/consensus-provider/consensus-provider.service.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ interface RequestRetryOptions {
 
 @Injectable()
 export class ConsensusProviderService {
-  protected rpcUrls: { main: string; backup: string };
+  protected rpcUrls: string[];
```
Comment (madlabman) [inline src/common/eth-providers/consensus-provider/consensus-provider.service.ts:34]: Will be nice to change to something like `apiUrls` in upcoming releases.

File: src/common/eth-providers/consensus-provider/consensus-provider.service.ts [hunk 2]
```diff
@@ -340,20 +337,27 @@ export class ConsensusProviderService {
...
-            throw e;
+    let res;
+    for (let i = 0; i < this.rpcUrls.length; i++) {
+      if (res) break;
+      res = await callback(this.rpcUrls[i])
+        .catch(rejectDelay(this.config.get('CL_BEACON_RPC_RETRY_DELAY_MS')))
+        .catch(() => retry(() => callback(this.rpcUrls[i])))
+        .catch((e: any) => {
+          if (fallbackConditionCallback(e)) {
+            if (!this.rpcUrls.length) {
```
Comment (madlabman) [inline src/common/eth-providers/consensus-provider/consensus-provider.service.ts:348]: `this.rpcUrls` will contain at least the first one endpoint, so the condition looks wrong.

File: src/common/eth-providers/execution-provider/execution-provider.module.ts [hunk 3]
```diff
@@ -3,14 +3,15 @@ import { FallbackProviderModule } from '@lido-nestjs/execution';
 import { PrometheusService, RequestStatus } from 'common/prometheus';
 import { ConfigService } from 'common/config';
 import { ExecutionProviderService } from './execution-provider.service';
+import { NonEmptyArray } from '@lido-nestjs/execution/dist/interfaces/non-empty-array';
```
Comment (madlabman) [inline src/common/eth-providers/execution-provider/execution-provider.module.ts:6]: Fill free to contribute to `@lido-nestjs` module to export used interface.

File: src/common/eth-providers/consensus-provider/consensus-provider.service.ts (hunk 1 above)
Comment (vgorkavenko) [reply src/common/eth-providers/consensus-provider/consensus-provider.service.ts]: Renamed all references

File: src/common/eth-providers/consensus-provider/consensus-provider.service.ts (hunk 2 above)
Comment (vgorkavenko) [reply src/common/eth-providers/consensus-provider/consensus-provider.service.ts]: fixed

File: src/common/eth-providers/execution-provider/execution-provider.module.ts (hunk 3 above)
Comment (vgorkavenko) [reply src/common/eth-providers/execution-provider/execution-provider.module.ts]: used another source

**Review by madlabman — APPROVED** [review]

**Pull request merged**